nixos/modules/services/matrix/conduwuit.nix

```diff
@@ -0,0 +1,299 @@
+  configFile = format.generate "conduwuit.toml" filteredSettings;
+in
+{
+  meta.maintainers = with lib.maintainers; [ niklaskorz ];
+  options.services.conduwuit = {
+    enable = lib.mkEnableOption "conduwuit";
+
+    user = lib.mkOption {
+      type = lib.types.nonEmptyStr;
+      description = ''
+        The user {command}`conduwuit` is run as.
+      '';
+      default = defaultUser;
+    };
+
+    group = lib.mkOption {
+      type = lib.types.nonEmptyStr;
+      description = ''
+        The group {command}`conduwuit` is run as.
+      '';
+      default = defaultGroup;
+    };
+
+    extraEnvironment = lib.mkOption {
+      type = lib.types.attrsOf lib.types.str;
+      description = "Extra Environment variables to pass to the conduwuit server.";
+      default = { };
+      example = {
+        RUST_BACKTRACE = "yes";
+      };
+    };
+
+    package = lib.mkPackageOption pkgs "conduwuit" { };
+
+    settings = lib.mkOption {
+      type = lib.types.submodule {
+        freeformType = format.type;
+        options = {
+          global.server_name = lib.mkOption {
+            type = lib.types.nonEmptyStr;
+            default = "";
+            example = "example.com";
+            description = "The server_name is the name of this server. It is used as a suffix for user # and room ids.";
+          };
+          global.address = lib.mkOption {
+            type = lib.types.nullOr (
+              lib.types.oneOf [
+                lib.types.nonEmptyStr
+                (lib.types.listOf lib.types.nonEmptyStr)
+              ]
+            );
+            default = null;
+            description = ''
+              Address (IPv4 or IPv6) to listen on for connections by the reverse proxy/tls terminator.
+              If set to `null`, conduwuit will listen on IPv4 and IPv6 localhost.
+              Must be `null` if `unix_socket_path` is set.
+
+              To listen on multiple addresses, specify a list. For example:
+              `[ "127.0.0.1" "::1" ]`
+            '';
+          };
+          global.port = lib.mkOption {
+            type = lib.types.oneOf [
+              lib.types.port
+              (lib.types.listOf lib.types.port)
+            ];
+            default = 6167;
+            description = "The port(s) conduwuit will be running on. You need to set up a reverse proxy in your web server (e.g. apache or nginx), so all requests to /_matrix on port 443 and 8448 will be forwarded to the conduwuit instance running on this port";
+          };
+          global.unix_socket_path = lib.mkOption {
+            type = lib.types.nullOr lib.types.path;
+            default = null;
+            description = ''
+              Listen on a UNIX socket at the specified path. If listening on a UNIX socket,
+              listening on an address will be disabled. The `address` option must be set to
+              `null` (the default value). You must add your reverse proxy to the `conduwuit`
```

Comment:
```suggestion
              Listen on a UNIX socket at the specified path.
              Listening on UNIX socket is exclusive with listening on addresses and must be disabled.
              You must add your reverse proxy to the `conduwuit`
```

Reply:
This just sounds more complicated without any extra useful information

Reply:
The 2nd and 3rd sentence are duplicated and can be combined that the two options are exclusive.

Reply:
The suggestion is grammatically incorrect and makes it harder to read.